lib/ci-helper.ts

```diff
@@ -62,6 +64,54 @@ export class CIHelper {
         this.urlRepo = `${this.urlBase}${this.config.repo.name}/`;
     }
 
+    public async setupGitHubAction(): Promise<void> {
+        // help dugite realize where `git` is...
+        process.env.LOCAL_GIT_DIRECTORY = "/usr/";
+        process.env.GIT_EXEC_PATH = "/usr/lib/git-core";
+
+        // get the access tokens via the inputs of the GitHub Action
+        this.setAccessToken("gitgitgadget", core.getInput("gitgitgadget-git-access-token"));
+        this.setAccessToken("git", core.getInput("git-git-access-token"));
+        this.setAccessToken("dscho", core.getInput("dscho-git-access-token"));
+
+        // eslint-disable-next-line security/detect-non-literal-fs-filename
+        if (!fs.existsSync(this.workDir)) await git(["init", "--bare", "--initial-branch", "main", this.workDir]);
+        for (const [key, value] of [
+            ["gc.auto", "0"],
+            ["remote.origin.url", `https://github.com/${this.config.repo.owner}/${this.config.repo.name}`],
```

Comment:
Using `this.config.repo.host` could help with off-github support. 

This applies in many places so it could be handled at another time.